torch_xla/_internal/rendezvous.py

```diff
@@ -44,4 +43,8 @@ def pjrt_rendezvous_handler(url: str,
             xr.process_count(),
             is_master=xr.process_index() == 0)
 
-  yield (_store, xr.global_ordinal(), world_size)
+  # In SPMD, the world size and rank are determined by the process count and
+  # index, while in multiprocess they are based on the device count and ordinal.
+  world_size = xr.process_count() if xr.is_spmd() else xr.world_size()
+  rank = xr.process_index() if xr.is_spmd() else xr.global_ordinal()
```

Comment:
Is the process idx local (per host)?

Reply:
Yeah, with SPMD we have a single process per host. process_index gives the global index of the current host's process in the runtime, so it will be different on each host.